osbs/build/user_params.py

```diff
@@ -288,22 +243,22 @@ def from_json(self, user_params_json):
             raise
         for key, value in json_dict.items():
             try:
-                self.convert_dict[key].value = value
-            except KeyError:
+                setattr(self, key, value)
+            except AttributeError:
                 continue
 
-    def set_if_exists(self, json_dict, param):
-        if self.convert_dict[param].value:
-            json_dict[param] = self.convert_dict[param].value
-
     def to_dict(self, keys):
         retdict = {}
         for key in keys:
-            self.set_if_exists(retdict, key)
+            value = getattr(self, key)
+            # TODO: why are we not including falsy values?
```

Comment:
Is there a good reason for this? Unit tests expect this behaviour, but it is strange.

Reply:
It's probably meant to not include `None` I cannot figure out why falsy values should be skiped here

Reply:
I think I'll leave it as it is for now, it worked fine all this time and, apparently, adding unnecessary things to the json output might be a bad idea.